DSNP-Wallet/Source/Common/Subscriptions/TransactionSubscriptionManager.swift

```diff
@@ -0,0 +1,88 @@
+//
+//  TransactionSubscriptionManager.swift
+//  DSNP-Wallet
+//
+//  Created by Ryan Sheh on 4/11/23.
+//
+
+import Foundation
+import SubstrateSdk
+import RobinHood
+
+typealias TransactionSubscriptionCompletion = ([TransactionSubscriptionResult])->()
+
+//EventSubscriptionManager relays event notifications to
+//whomever inherits this delegate
+protocol TransactionHandlerDelegate: AnyObject {
+    func handleTransactions(result: Result<[DataProviderChange<TransactionHistoryItem>], Error>)
+}
+
+protocol TransactionSubscriptionManagerProtocol {
+    func process(blockHash: Data, completion: @escaping TransactionSubscriptionCompletion)
+    func subscribeToTransaction(with txHashString: String) 
+}
+
+//MARK: Wrapper class for TransactionSubscription and TransactionProvider
+// To handle System->Events following an Extrinsic
+class TransactionSubscriptionManager: TransactionSubscriptionManagerProtocol, TransactionLocalStorageSubscriber,
+                                      TransactionLocalSubscriptionHandler {
+    //MARK: Transaction Subscription
+    var transactionSubscription: TransactionSubscription
+    weak var handlerDelegate: TransactionHandlerDelegate?
+    
+    //MARK: TransactionProvider
+    var transactionLocalSubscriptionFactory: TransactionLocalSubscriptionFactoryProtocol
+    var transactionLocalSubscriptionHandler: TransactionLocalSubscriptionHandler?
+    private var transactionProvider: StreamableProvider<TransactionHistoryItem>?
+    
+    init(transactionLocalSubscriptionFactory: TransactionLocalSubscriptionFactoryProtocol,
+         chainRegistry: ChainRegistryProtocol,
+         repositoryFactory: SubstrateRepositoryFactoryProtocol,
+         accountId: AccountId,
+         chainModel: ChainModel,
+         storageRequestFactory: StorageRequestFactoryProtocol,
+         eventCenter: EventCenterProtocol,
+         handlerDelegate: TransactionHandlerDelegate?,
+         logger: LoggerProtocol
+    ) {
+        self.transactionLocalSubscriptionFactory = transactionLocalSubscriptionFactory
+        self.handlerDelegate = handlerDelegate
+        
+        let address = try! accountId.toAddress(using: chainModel.chainFormat)
```

Comment:
in general I like wrapping these in a do catch block, in case the `try` fails, we save a crash